### ISSUE TITLE
Minimap should never overlap text in open buffer in absolute mode

### ISSUE THREAD

**Issue by mnquintana** [atom-minimap/minimap#344]
I'm loving the new absolute mode, but when I originally submitted https://github.com/atom-minimap/minimap/issues/337, I had been thinking the minimap would have an opaque background, so:
![screenshot 2015-06-11 07 23 53](https://cloud.githubusercontent.com/assets/823545/8105879/e10a91d6-100a-11e5-87ff-139bf118839d.png)

Instead of:
![screenshot 2015-06-11 07 22 25](https://cloud.githubusercontent.com/assets/823545/8105881/e55db394-100a-11e5-9de8-0e7fa94668dc.png)

That said, I'm not terribly opposed to the current styling, but it is kinda hard to read text behind the minimap, and I could see it being worse with other syntax themes.

**Comment by abe33** [comment]
Make sense, I'll see how I can do that. Ideally you want to only have an opaque background behind the rendered lines and not the full minimap, which mean I'll have to make some additional draw, the worst would be to make the whole minimap opaque as it would hide end of lines even when the minimap's content doesn't occupy the full height.
That make me think that I probably should find a way to expand the the text-editor width when in absolute mode, because you won't be able to scroll all the way to the right as the minimap will always overlay the end of line. If you have any suggestions on how to achieve such a thing without breaking how the editor behave? I'm not sure about using some right padding.

**Comment by mnquintana** [comment]
I think it would probably be easier to adjust the height of the minimap in absolute mode to fit the `canvas` height. Might have to write a function for that though – I'm not sure if it's possible with pure CSS?

**Comment by polarathene** [comment]
@abe33 Would like to see this enhancement, having a setting to configure a background fill colour/opacity for the minimap would be useful too when not in absolute mode. The theme I'm using has no distinction between the editor background and where the minimap area begins, when not in absolute mode text just disappears and it may not always be evident that it continues unless I notice the small horizontal scroll bar. Alternatively a border/separator line would help.

**Comment by fundon** [comment]
If you want that you can use custom the background-color of minimap.

https://github.com/atom-minimap/minimap#changing-the-minimaps-background

```css
atom-text-editor atom-text-editor-minimap,
atom-text-editor::shadow atom-text-editor-minimap {
  background: rgb(40, 44, 52);
}
```

**Comment by fundon** [comment]
Closed.

You can re-open it if the issue is still here.

**Comment by mnquintana** [comment]
Whoops, sorry @fundon – this suggestion changed a bit from my initial post. What I realized I was really asking for was to expand the text editor width in absolute mode to make it so the minimap never overlaps an open buffer. So just changing the background color manually doesn't really fit this use case. Would you mind reopening? :smile:

**Comment by e7d** [comment]
I got myself in the same situation as @mnquintana.  

I found a solution that suits for me, but only because i only use the "absolute mode. The idea is to add a "gutter" on the right, to let the "text-editor" breathe. I achieved it as following, editing the `minimap.less` file:
``` less
atom-text-editor::shadow, atom-text-editor, html {
  .scroll-view {
    margin-right: 10%;
  }
}
```

The margin is set to 10%, since it is the width set for the `atom-text-editor-minimap` tag.

The main issue is the margin will apply in both modes, absolute or not. It should only be there in absolute mode.
I could not achieve it, since the `.absolute` class is set on the `atom-text-editor-minimap` which is not a parent of the "text editor" tag I modified.

Without setting the `.absolute' class at an upper level, I don't see how to achieve the expected behaviour.
